### ISSUE TITLE
Bash Completion Fails

### ISSUE THREAD

**Issue by robml** [pdm-project/pdm#954]
- [X] I have searched the issue tracker and believe that this is not a duplicate.

**Make sure you run commands with `-v` flag before pasting the output.**

## Steps to reproduce

On trying to execute this command:

`sudo pdm completion bash > /etc/bash_completion.d/pdm.bash-completion`

I get the output below

## Actual behavior

`-bash: /etc/bash_completion.d/pdm.bash-completion: Permission denied`

## Expected behavior

I searched up online and found a mention in `poetry` and tried adding the following in my `.bashrc` file:

`. <(pdm completions bash)`

Now, I am no bash expert, and wonder whether that line does anything or if there is another workaround, because every time I load up the shell all that happens is it prints the `pdm` USAGE message.

## Environment Information

Everything else works fine so far.

```bash
# Paste the output of `pdm info && pdm info --env` below:

```

**Comment by pawamoy** [comment]
The issue with redirections is that they are part of the current shell, which does not have root permissions.

Instead of

```
sudo pdm completion bash > /etc/bash_completion.d/pdm.bash-completion
```

try

```
sudo bash -c 'pdm completion bash > /etc/bash_completion.d/pdm.bash-completion'
```

This way the redirection itself will be able to write to /etc.

**Comment by robml** [comment]
> The issue with redirections is that they are part of the current shell, which does not have root permissions.
> 
> Instead of
> 
> ```
> sudo pdm completion bash > /etc/bash_completion.d/pdm.bash-completion
> ```
> 
> try
> 
> ```
> sudo bash -c 'pdm completion bash > /etc/bash_completion.d/pdm.bash-completion'
> ```
> 
> This way the redirection itself will be able to write to /etc.

Thanks for the tip, however running it and even trying from `root` directly I noticed that I would have to install `pdm` on there for it to work since I get this message `bash: pdm: command not found`

I decided a workaround was to simply copy the message from `pdm completion bash` and paste it directly into the file `/etc/bash_completion.d/pdm.bash-completion` while logged in as `root`.

Not an elegant solution by any means but seems to be fine, the previous line I menttioned (`. <(pdm completions bash)`) I have no idea if it was doing anything and just commented it out. Finally (for anyone else reading) I noticed that `pdm` was the only program using my `.bash_profile` which overrides my `.profile` so had to add two lines to source it for the whole system to work seamlessly.
